apprating/src/main/java/com/mindera/skeletoid/apprating/AppRatingInitializer.kt

```diff
@@ -0,0 +1,71 @@
+package com.mindera.skeletoid.apprating
+
+import android.content.Context
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogCallback
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponse
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponseCallback
+import com.mindera.skeletoid.apprating.controller.AppRatingController
+
+/**
+ * Entry point to setup and call the app rating dialog
+ */
+object AppRatingInitializer {
+
+    private val controller: AppRatingController by lazy {
+        AppRatingController()
+    }
+
+    private lateinit var callback: AppRatingDialogCallback
+
+    /**
+     * Sets up the conditions to prompt the dialog.
+     * If it's not called, promptDialog will prompt the rating dialog immediately.
+     *
+     * @param countsPerTimeInterval Pair<Int, Long> values with the maximum number of times the dialog can be prompt per time range (in days)
+     * @param promptTimeInterval Time distance between prompts (in days)
+     */
+    fun init(
+        context: Context,
+        callback: AppRatingDialogCallback,
+        countsPerTimeInterval: Pair<Int, Long> = Pair(
+            context.resources.getInteger(R.integer.maximum_counts),
+            context.resources.getInteger(R.integer.maximum_counts_time_interval).toLong()
+        ),
+        promptTimeInterval: Long = context.resources.getInteger(R.integer.minimum_time_between_prompt).toLong()
+    ) {
+        this.callback = callback
+        controller.setupConditions(countsPerTimeInterval, promptTimeInterval)
```

Comment:
shouldn't we initialise here a job to schedule a dialog prompt based on time interval?

Reply:
The job schedule is only triggered when the user clicks "rate later"

Reply:
But don't we want to trigger a time driven app rating from the moment we init this component?

Reply:
I think we don't. The dialog should only be prompted after the action of creating an order. The time driven prompt is only used depending on the user response to the first prompt

Reply:
So if the user don't make any order in the app we never prompt them?

Reply:
According to overflow, this is correct
> The dialog should only be prompted after the action of creating an order. 

Just thinking if we want to add the possibility to whoever might want to use this feature to 
> trigger a time driven app rating dialog prompt from the moment we init this component

What do you think?

Reply:
Yap, i think we could pass a flag here and have a time driven prompt based on the value passed in the `promptTimeInterval`. Do you agree?